components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/ImageVectorFileSpec.kt

```diff
@@ -99,41 +88,30 @@ internal class ImageVectorFileSpec(private val config: ImageVectorSpecConfig) {
     private fun iconProperty(
         vector: Vector,
         iconPackClassName: ClassName?,
-        backingProperty: PropertySpec,
     ): PropertySpec = propertySpecBuilder(name = config.iconName, type = ClassNames.ImageVector) {
         receiver(iconPackClassName)
-        getter(iconFun(vector = vector, backingProperty = backingProperty))
-    }
-
-    private fun iconFun(vector: Vector, backingProperty: PropertySpec): FunSpec {
-        return getterFunSpecBuilder {
-            addCode(
-                buildCodeBlock {
-                    beginControlFlow("if (%N != null)", backingProperty)
-                    addStatement("return %N!!", backingProperty)
-                    endControlFlow()
-                },
-            )
-            addCode(
-                buildCodeBlock {
-                    addCode("%N = ", backingProperty)
-                    add(
-                        imageVectorBuilderSpecs(
-                            iconName = when {
-                                config.iconNestedPack.isEmpty() -> config.iconName
-                                else -> "${config.iconNestedPack}.${config.iconName}"
-                            },
-                            vector = vector,
-                            path = {
-                                vector.nodes.forEach { node -> addVectorNode(node) }
-                            },
-                        ),
-                    )
-                },
+        val codeBlock = buildCodeBlock {
+            add(
+                imageVectorBuilderSpecs(
+                    iconName = when {
+                        config.iconNestedPack.isEmpty() -> config.iconName
+                        else -> "${config.iconNestedPack}.${config.iconName}"
+                    },
+                    vector = vector,
+                    path = {
+                        vector.nodes.forEach { node -> addVectorNode(node) }
+                    },
+                ),
             )
-            addStatement("")
-            addStatement("return %N!!", backingProperty)
         }
+
+        delegate(
+            CodeBlock.builder()
+                .beginControlFlow("lazy(%T.NONE)", ClassNames.LazyThreadSafetyMode)
+                .add(codeBlock)
+                .endControlFlow()
+                .build(),
+        )
```

Comment:
All here, you can copy and paste anytime. I'll leave the PR here until I have more time to take or finish by you.